### PR TITLE
Fix startup penalty on IO thread health

### PR DESCRIPTION
**Pull request by andyp1per**
I occasionally get these IO thread died errors when doing development and on further investigation it looks like a transient startup condition from which we never recover. Checking the logging thread every second during normal operation is fine, but at startup there is a whole pile of intense CPU load going on that can catch the thread including expected delays. And once its gone you can't get it back. This simply checks that we are not in an expected delay and raises the timeout to match that of the file logger.

### CODE REVIEW

File: libraries/AP_Logger/AP_Logger_Block.cpp [hunk 1]
```diff
@@ -822,10 +822,13 @@ bool AP_Logger_Block::logging_failed() const
...
 
+// detect whether the IO thread is running, since this is considered a catastrophic failure for the logging system
+// better be really, really sure
 bool AP_Logger_Block::io_thread_alive() const
 {
-    // if the io thread hasn't had a heartbeat in 1s it is dead
-    return (AP_HAL::millis() - io_timer_heartbeat) < 1000U;
+    uint32_t now = AP_HAL::millis();
+    // if the io thread hasn't had a heartbeat in 1s and its some time after startup it is dead
+    return now > 10000U && (now - io_timer_heartbeat) < 1000U;
```
Comment (amilcarlucas) [inline libraries/AP_Logger/AP_Logger_Block.cpp]: This is always false for the first 10 seconds. Is that what you want?
It seams to go against the explanation on the PR.

Comment (andyp1per) [reply libraries/AP_Logger/AP_Logger_Block.cpp]: Yeah I just realized - now corrected

**Comment by andyp1per** [comment]
More investigation required, this is not having the desired effect.

**Comment by andyp1per** [comment]
@magicrub - how so? It's the same on the SD card logger. My thinking is that once its gone its gone and waiting an extra couple of seconds to be sure should be ok. It doesn't affect flight control.

**Review by tridge — APPROVED** [review]

**Pull request merged**